cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/assumptions/Assumption.kt

```diff
@@ -322,6 +322,15 @@ interface HasAssumptions {
         return this
     }
 
+    /**
+     * This is function is used to collect all assumptions stored in the [HasAssumptions] object and
+     * its contained objects that can have assumptions. While this default implementation is simple,
+     * composite [HasAssumptions] objects should return the assumptions of the contained objects.
+     */
+    fun collectAssumptions(): Set<Assumption> {
+        return assumptions.toSet()
```

Comment:
Do we need some loop detection in the implementations of this method? We seem to traverse some sub-graphs sometimes, which could result in problems, can't it? Currently, probably not a big deal though

Reply:
That could in theory happen if for example a QueryTree contains itself as subtree, or a composite result contains itself as a result. However, that is only possible if a result is dependent on itself, which clearly a problem in the implementation, not a case that we have to handle during runtime. I would, therefore, not implement a loop detection or worklist. I could be convince do consider a sanity check in the future.